### PR TITLE
Create component for Connected Site and Permissions

### PR DESCRIPTION
**Pull request by vthomas13**
## Explanation
This PR creates a new component, ConnectedSitePermissionsModal. This component is to be implemented in future PRs related to the Connected Sites flow.

## Screenshots/Screencaps

image
image

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/multichain/connected-site-permissions-modal/connected-site-permissions-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+} from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+
+interface ConnectedSitePermissionsModalProps {
+  onClose: () => void;
+  siteIcon: string;
+  siteName: string;
+}
+
+export const ConnectedSitePermissionsModal: FC<
```
Comment (DDDDDanica) [inline ui/components/multichain/connected-site-permissions-modal/connected-site-permissions-modal.tsx:36]: Can we have simple test for this component?

File: ui/components/multichain/connected-site-permissions-pill/connected-site-permissions-pill.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  siteName: string;
+  siteIcon: string;
+}
+
+export const ConnectedSitePermissionsPill: FC<
+  ConnectedSitePermissionsPillProps
+> = ({ actionButtonLabel, siteName, siteIcon }) => {
+  const dispatch = useDispatch();
+  return (
+    <TagUrl
```
Comment (DDDDDanica) [inline ui/components/multichain/connected-site-permissions-pill/connected-site-permissions-pill.tsx:17]: If there's only 1 component from existing component library, can we directly do 
```
<Box
            width={BlockSize.TenTwelfths}
            justifyContent={JustifyContent.center}
          >
            <TagUrl
              /// other props
              siteName={siteName}
              siteIcon={siteIcon}
            />
          </Box>
```

File: ui/store/actions.ts [hunk 3]
```diff
@@ -2427,6 +2427,19 @@ export function hideIpfsModal(): Action {
...
+    type: actionConstants.SHOW_SITE_PERMISSIONS_MODAL,
+  };
+}
+
+export function hideSitePermissionsModal(): Action {
+  return {
+    type: actionConstants.HIDE_SITE_PERMISSIONS_MODAL,
+  };
+}
+
```
Comment (NidhiKJha) [inline ui/store/actions.ts:2442]: Do you think we can move this to local state in modal?

File: ui/components/multichain/connected-site-permissions-modal/connected-site-permissions-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+> = ({ onClose, siteIcon, siteName }) => {
+  const [open, setOpen] = useState(true);
+  const t = useI18nContext();
+
+  const handleOnClose = () => {
+    setOpen(false);
+    onClose();
+  };
+  return (
+    <Modal isOpen={open} onClose={handleOnClose}>
```
Comment (NidhiKJha) [inline ui/components/multichain/connected-site-permissions-modal/connected-site-permissions-modal.tsx:47]: I think we should add a className or data-test-id here.

**Review by NidhiKJha — COMMENTED** [review]
Great Start! I think these components should be moved inside the connections component in multichain, since we want to render this on Connections Page. Wdyt?

Also, can we add tests to these components?

**Comment by darkwing** [comment]
@vthomas13 Where are we with this PR?

**Comment by vthomas13** [comment]
> @vthomas13 Where are we with this PR?

Hey, I think I lost track of this one, but I'm working on the feedback today.

File: ui/store/actions.ts (hunk 3 above)
Comment (darkwing) [reply ui/store/actions.ts:2456]: Echoing Nidhi's question here -- can showing and hiding this functionality be done by local state (aka `useState`) of a parent component, or can this modal be launched from multiple places in the app?

Comment (vthomas13) [reply ui/store/actions.ts]: Added [5f4be6f](https://github.com/MetaMask/metamask-extension/pull/20633/commits/5f4be6fa1abef51907b06bb264da7f25bb2642bc)

File: ui/components/multichain/connected-site-permissions-modal/connected-site-permissions-modal.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  onClose: () => void;
+  siteName: string;
+  siteIcon: string;
+}
+
+export default {
+  title: 'Components/Multichain/ConnectedSitePermissionsModal',
+  component: ConnectedSitePermissionsModal,
+  argTypes: {
+    onClose: { control: 'function', action: 'onClose' },
```
Comment (georgewrmarshall) [inline ui/components/multichain/connected-site-permissions-modal/connected-site-permissions-modal.stories.tsx:15]: I don't think you need the function control here if you have the action. It assumes the type is a funciton
```suggestion
    onClose: { action: 'onClose' },
```

File: ui/components/multichain/connected-site-permissions-modal/index.scss [hunk 6]
```diff
@@ -0,0 +1,4 @@
+.connected-site-permissions-pill {
+  flex: 0 0 auto;
+  margin: auto;
```
Comment (NidhiKJha) [inline ui/components/multichain/connected-site-permissions-modal/index.scss:3]: I think we can remove the scss here and use the utility Box Props to style this component. wdyt?

File: ui/components/multichain/connected-site-permissions-modal/connected-site-permissions-modal.tsx [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+        <Box
+          display={Display.Flex}
+          flexDirection={FlexDirection.Column}
+          alignItems={AlignItems.center}
+          paddingTop={6}
+          paddingBottom={2}
+          gap={4}
+        >
+          <Box
+            width={BlockSize.TenTwelfths}
```
Comment (NidhiKJha) [inline ui/components/multichain/connected-site-permissions-modal/connected-site-permissions-modal.tsx:67]: Hmm....why are we using this width? 🤔

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
This is looking really great @vthomas13. Just left some comments

**Comment by vthomas13** [comment]
Closing this PR due to design change, @darkwing

**Pull request closed without merging**